test/antrea/k8spolicy.dl

```diff
@@ -0,0 +1,346 @@
+	// traffic only if the traffic matches at least one port in the list.
+	// +optional
+	ports: Vec<NetworkPolicyPort>,
+
+	// List of sources which should be able to access the pods selected for this rule.
+	// Items in this list are combined using a logical OR operation. If this field is
+	// empty or missing, this rule matches all sources (traffic not restricted by
+	// source). If this field is present and contains at least on item, this rule
+	// allows traffic only if the traffic matches at least one item in the from list.
+	// +optional
+	from: Vec<NetworkPolicyPeer>
+}
+
+// NetworkPolicyEgressRule describes a particular set of traffic that is allowed out of pods
+// matched by a NetworkPolicySpec's podSelector. The traffic must match both ports and to.
+// This type is beta-level in 1.8
+typedef NetworkPolicyEgressRule = NetworkPolicyEgressRule {
+	// List of destination ports for outgoing traffic.
+	// Each item in this list is combined using a logical OR. If this field is
+	// empty or missing, this rule matches all ports (traffic not restricted by port).
+	// If this field is present and contains at least one item, then this rule allows
+	// traffic only if the traffic matches at least one port in the list.
+	// +optional
+	ports: Vec<NetworkPolicyPort>,
+	// List of destinations for outgoing traffic of pods selected for this rule.
+	// Items in this list are combined using a logical OR operation. If this field is
+	// empty or missing, this rule matches all destinations (traffic not restricted by
+	// destination). If this field is present and contains at least one item, this rule
+	// allows traffic only if the traffic matches at least one item in the to list.
+	// +optional
+	to: Vec<NetworkPolicyPeer>
+}
+
+typedef Protocol = IString
+
+typedef IntOrString = Either<signed<32>, string>
+
+function intOrStringIntValue(x: IntOrString): signed<32> =
+{
+    match (x) {
+        Left{i} -> i,
+        Right{s} -> {
+            match (parse_dec_i64(s)) {
+                None -> 0,
+                Some{i} -> i as signed<32>
+            }
+        }
+	}
+}
+
+// NetworkPolicyPort describes a port to allow traffic on
+typedef NetworkPolicyPort = NetworkPolicyPort {
+	// The protocol (TCP, UDP, or SCTP) which traffic must match. If not specified, this
+	// field defaults to TCP.
+	// +optional
+	protocol: Option<Protocol>,
+
+	// The port on the given protocol. This can either be a numerical or named port on
+	// a pod. If this field is not provided, this matches all port names and numbers.
+	// +optional
+	port: Option<IntOrString>
+}
+
+// IPBlock describes a particular CIDR (Ex. "192.168.1.1/24") that is allowed to the pods
+// matched by a NetworkPolicySpec's podSelector. The except entry describes CIDRs that should
+// not be included within this rule.
+typedef IPBlock = IPBlock {
+	// CIDR is a string representing the IP Block
+	// Valid examples are "192.168.1.1/24"
+	cidr: string,
```

Comment:
again strings... I wonder whether if you could have an enum IPCblock and and then a variant named IPBlockString.